plugins/woocommerce/src/Internal/BatchProcessing/BatchProcessingController.php

```diff
@@ -75,6 +80,13 @@ function ( $batch_process ) {
 			2
 		);
 
+		add_action(
+			'shutdown',
+			function () {
+				$this->remove_or_retry_failed_processors();
+			}
+		);
```

Comment:
Well... in theory this could be tied to any other hook, it doesn't have to be the `shutdown` one specifically, but whatever we use I believe it'll still have to run on every init/shutdown/etc.

What I'm trying to do here is bring back to life (or completely remove, depending on the number of attempts) processors that have failed due to serious errors such as timeouts, syntax or fatal errors.
For exceptions [we have some handling](https://github.com/woocommerce/woocommerce/blob/e281fec70a33cd75dd22c5aedf065a7618be8b99/plugins/woocommerce/src/Internal/BatchProcessing/BatchProcessingController.php#L169) because we can just capture them, but for serious errors, we can't do much from within `process_next_batch_for_single_processor_core()` as the underlying A-S action has probably been killed and well... execution has stopped, leaving the processor in limbo: enqueued but not really scheduled (and so it won't ever run).

The code in `shutdown` specifically looks at those enqueued but not scheduled processors and either schedules them again or removes them.

I'm obviously open to other ways to do this or hooks to use!